### DISCUSSION TITLE
[Modal] Approve behaviour

### DISCUSSION THREAD

**Discussion by jamessampford** [fomantic/Fomantic-UI#1263] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
I don't believe this exists, or at least not mentioned in the docs... Am wondering if perhaps there should be an "approve" behaviour added to the modal so custom keyboard handlers could be used to approve the modal, as opposed to needing to click on approve
`$('.ui.modal').modal('approve');`

May also want to add "deny" as well for the same reason (yes, can already hide but may want to trigger onDeny as well)
`$('.ui.modal').modal('deny');`

Another example usage may be an auto-logout prompt that may cancel a time when the mouse is moved and "denies" the prompt and cancels the auto-logout event (or potentially approves it and follows the logout procedure)

**Comment by lubber-de** [comment, marked as answer]
This should already be possible.
```javascript
$('.ui.modal').modal('event approve');
$('.ui.modal').modal('event deny');
```
The events are meant to be used internally only (thus not documented), but the behavior system is able to call any existing method inside the module

**Comment by jamessampford** [comment]
Aha!

I did try the below, so close but no cigar...
`$('.ui.modal').modal('event', 'approve');`

I wonder if maybe simply having an approve/deny behaviour would be better to have since internals can of course change over time (and would be more standardised)

**Comment by ko2in** [comment]
Close since @lubber-de already provided the solution. Please feel free to reopen or create a separate issue if you've feature request about this.
